var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1269,6 +1268,9 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         """Set PYTHONPATH to include the site-packages directory for the
         extension and any other python extensions it depends on.
         """
+        if sys.platform == "win32":
+            return
```

Comment:
why's this here?

Reply:
Cause all the logic below is not for Windows (setting the `compiler-wrapper`, setting `LDSHARED` and `LDCXXSHARED`).

Reply:
See https://github.com/spack/spack/pull/45189/commits/0cae1bf29297f3456b8fad3fed5bc108f5372fc7